crates/executor/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use std::future::Future;
+use std::mem;
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll, Wake, Waker};
+
+use wasi::io;
+
+wit_bindgen::generate!({
+    world: "imports",
+    path: "io.wit",
+});
+
+impl std::fmt::Display for io::streams::Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.to_debug_string())
+    }
+}
+
+impl std::error::Error for io::streams::Error {}
+
+pub static WAKERS: Mutex<Vec<(io::poll::Pollable, Waker)>> = Mutex::new(Vec::new());
```

Comment:
Rather than make this public, could we perhaps add a `pub fn push_waker(pollable: Pollable, waker: Waker)` so apps can call that instead?  I'm thinking it would be better to expose as little of the implementation as possible in the public API.

Reply:
Great idea. I wasn't very comfortable with this `pub static` to begin with.